src/ccl_cls.c

```diff
@@ -1067,18 +848,19 @@ static void get_k_interval(ccl_cosmology *cosmo,CCL_ClWorkspace *w,
 			   CCL_ClTracer *clt1,CCL_ClTracer *clt2,int l,
 			   double *lkmin,double *lkmax)
 {
-  double chimin,chimax;
-  int cut_low_1=0,cut_low_2=0;
-
-  //Define a minimum distance only if no lensing is needed
-  if((clt1->tracer_type==CL_TRACER_NC) && (clt1->has_magnification==0)) cut_low_1=1;
-  if((clt2->tracer_type==CL_TRACER_NC) && (clt2->has_magnification==0)) cut_low_2=1;
-
   if(l<w->l_limber) {
-    chimin=2*(l+0.5)/ccl_splines->K_MAX;
-    chimax=0.5*(l+0.5)/ccl_splines->K_MIN;
+    //If non-Limber, we need to integrate over the whole range of k.
+    *lkmin=log10(ccl_splines->K_MIN);
+    *lkmax=log10(ccl_splines->K_MAX);
```

Comment:
Is this bit still needed?  You got rid of most of the `l < w_limber` stuff.  But maybe this is still necessary even when doing angpow for this part of the range?